reference/src/representation/unions.md

```diff
@@ -0,0 +1,36 @@
+# Representation of unions
+
+**Disclaimer:** This chapter represents the consensus from issue
+[#13].  The statements in here are not (yet) "guaranteed"
+not to change until an RFC ratifies them.
+
+[#13]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/13
+
+The only degree of freedom the compiler has when computing the layout of a union
+like
+
+```rust
+union U { f1: T1, f2: T2 }
+```
+
+is to determine the offset of the fields.  The layout of these fields themselves
+is already entirely determined by their types, and since we intend to allow
+creating references to fields (`&u.f1`), unions do not have any wiggle-room
+there.
+
```

Comment:
Please include a note that unions may contain values that do not correspond to a valid value of any of their fields.

Reply:
That's not a layout question ("where do which fields go, what about alignment and size?"), but a validity question ("which bit patterns are allowed?"). It seems in the wrong place here.

Reply:
It seems like a "representation" question. It doesn't have to go in *this* paragraph, but it seems like it should go somewhere in this document.

I'm open to suggestions if it should appear elsewhere instead, though.

Reply:
I agree with @RalfJung that seems like an *invariant* question more than layout. I view layout as answering "when you go to write/read this field, at what offset will you find it?" whereas this question is sort of "what might you find there". 

In particular, I think layout+representation are purely static properties.

Reply:
This can be resolved at a later date during editing--for the purposes of this PR, validity is out of scope.